arangod/StorageEngine/TransactionState.cpp

```diff
@@ -120,6 +125,36 @@ TransactionState::Cookie::ptr TransactionState::cookie(void const* key,
 /// @brief add a collection to a transaction
 Result TransactionState::addCollection(DataSourceId cid, std::string const& cname,
                                        AccessMode::Type accessType, bool lockUsage) {
+#ifdef ARANGODB_ENABLE_FAILURE_TESTS
+  TRI_IF_FAILURE(("WaitOnLock::" + cname).c_str()) {
+    auto& raceController = basics::DebugRaceController::sharedInstance();
+    if (!raceController.didTrigger()) {
+      raceController.waitForOthers(2, _id, vocbase().server());
+      // Slice out the first char, then we have a number
+      uint32_t shardNum = basics::StringUtils::uint32(&cname.back(), 1);
+      std::vector<std::any> const& data = raceController.data();
```

Comment:
`data()` now returns a copy, and to make this more clear at the call site, we should change this to:
```suggestion
      std::vector<std::any> const data = raceController.data();
```